modules/MusicDownloader.js

```diff
@@ -59,24 +91,33 @@ async function downloadPlaylist(url) {
         clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
     });
 
-    const data = await spotify.getPlaylist(url);
-    for (let i in data.tracks) {
-        let song = await getTrackInfo(data.tracks[i])
-        logger('log',`Pobieranie: ${song.name+' by: '+song.artists.join(', ')}`,'downloadPlaylist');
-    }
-    const playlist = await spotify.downloadPlaylist(url)
-    for (let i in playlist) {
-        let song = await getTrackInfo(data.tracks[i])
-        const dir = sterylizator(data.name);
-        if (!fs.existsSync(`./mp3/onDemand/${dir}`)){
-            fs.mkdirSync(`./mp3/onDemand/${dir}`);
+    try {
+        const data = await spotify.getPlaylist(url);
+        for (let i in data.tracks) {
+            let song = await getTrackInfo(data.tracks[i])
+            logger('log', `Pobieranie: ${song.name + ' by: ' + song.artists.join(', ')}`, 'downloadPlaylist');
         }
-        const file = sterylizator(song.artists.join('-')+'_'+song.name);
-        if (fs.existsSync(`./mp3/onDemand/${dir}/${file}.mp3`)) {
-            logger('warn',`${file}.mp3 - Plik istnieje!`,'downloadPlaylist');
-            continue;
+        const playlist = await spotify.downloadPlaylist(url)
+        for (let i in playlist) {
+            let song = await getTrackInfo(data.tracks[i])
+            const dir = sterylizator(data.name);
+            if (!fs.existsSync(`./mp3/onDemand/${dir}`)) {
+                fs.mkdirSync(`./mp3/onDemand/${dir}`);
+            }
+            const file = sterylizator(song.artists.join('-') + '_' + song.name);
+            if (fs.existsSync(`./mp3/onDemand/${dir}/${file}.mp3`)) {
+                logger('warn', `${file}.mp3 - Plik istnieje!`, 'downloadPlaylist');
+                continue;
+            }
+            fs.writeFileSync(`./mp3/onDemand/${dir}/${file}.mp3`, playlist[i]);
+        }
+    } catch (e) {
+        logger('error', "Błąd w trakcie wykonywania funkcji downloadSong", 'downloadPlaylist');
+        logger('error', e, 'downloadPlaylist');
+        if (global.debugmode === true) {
+            DebugSaveToFile('MusicDownloader', 'downloadPlaylist', 'error_main_function', e);
```

Comment:
Zmień z error_main_function na catched_error
```suggestion
            DebugSaveToFile('MusicDownloader', 'downloadPlaylist', 'catched_error', e);
```